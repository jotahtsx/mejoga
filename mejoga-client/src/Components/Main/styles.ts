'use client'

import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100vh;
  background-color: #06092b;
`

export const Title = styled.h1`
  font-size: 2rem;
  margin-bottom: 10px;
`

export const Description = styled.p`
  font-size: 1rem;
`
