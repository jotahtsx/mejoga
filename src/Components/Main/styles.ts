'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-color: #06092b;
`

export const Title = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
