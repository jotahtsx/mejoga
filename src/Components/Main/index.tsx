'use client'

import { Wrapper } from './styles'

type MainProps = {
  title: string
}

const Main = ({ title }: MainProps) => {
  return (
    <Wrapper>
      <h1>{title}</h1>
    </Wrapper>
  )
}

export default Main
