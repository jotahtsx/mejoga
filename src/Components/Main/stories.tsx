import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import Main from './index'

const meta: Meta<typeof Main> = {
  title: 'Main',
  component: Main,
}

export default meta

const Template: StoryFn<typeof Main> = (args) => <Main {...args} />

export const Default = Template.bind({})
Default.args = {
  title: 'TÃ­tulo do Storybook', // Aqui estamos passando a prop title
}
