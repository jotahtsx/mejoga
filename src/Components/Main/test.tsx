import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Main from './index'

describe('<Main />', () => {
  it('Precisa renderizar o header corretamente e gerar o snapshot', () => {
    // Renderiza o componente com a prop title
    const { container } = render(<Main title="Me Joga" />)

    // Verifica se o título correto está na tela
    expect(screen.getByRole('heading', { name: /me joga/i })).toBeInTheDocument()

    // Gera o snapshot para comparação visual
    expect(container.firstChild).toMatchSnapshot()
  })

  it('As cores precisam estar no padrão da verificação', () => {
    // Renderiza o componente com a prop title
    const { container } = render(<Main title="Me Joga" />)

    // Verifica se a cor de fundo está correta
    expect(container.firstChild).toHaveStyle('background-color: rgb(6, 9, 43)')
  })
})
