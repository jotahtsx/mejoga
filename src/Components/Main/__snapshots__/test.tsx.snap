// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Precisa renderizar o header corretamente e gerar o snapshot 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-color: #06092b;
}

<main
  class="c0"
>
  <h1>
    Me Joga
  </h1>
</main>
`;
